tests/test_tensorflow_data_api.py

```diff
@@ -205,21 +247,23 @@ def test_dataset_generator_batch_output(
                 TensorflowTileDBDenseDataset._generator(
                     x=x,
                     y=y,
+                    batch_size=BATCH_SIZE,
+                    batch_shuffle=batch_shuffle,
+                    within_batch_shuffle=within_batch_shuffle,
                     x_attribute_names=attribute_names,
                     y_attribute_names=attribute_names,
                     rows=ROWS,
-                    batch_size=BATCH_SIZE,
                 )
             )
 
             assert len(generated_data) == 2 * num_of_attributes
 
             for attr in range(num_of_attributes):
-                assert generated_data[attr].shape == (
+                assert generated_data[attr].shape <= (
```

Comment:
Why does this condition change?

Reply:
Because of batch shuffling it might hit the last batch, which could be smaller than the batch size.